libsrc/c64/crt0.s

```diff
@@ -39,35 +38,30 @@ L1:     lda     sp,x
 ; Switch off the BASIC ROM.
 
         lda     $01
-        pha                     ; Remember the value
+        sta     mmusave         ; Save the memory configuration
         and     #$F8
         ora     #$06            ; Enable Kernal+I/O, disable BASIC
         sta     $01
 
-; Clear the BSS data.
-
-        jsr     zerobss
-
-; Save some system settings; and, set up the stack.
-
-        pla
-        sta     mmusave         ; Save the memory configuration
-
         tsx
         stx     spsave          ; Save the system stack ptr
 
-        lda     #<(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
-        sta     sp
-        lda     #>(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
-        sta     sp+1            ; Set argument stack ptr
+; Allow some re-entrancy by skipping the next task if it already was done.
+; This often can let us rerun the program without reloading it.
```

Comment:
I very often rerun programs.  I will load a program, run it on a directory/disk, then change to another directory/disk, rerun, then change yet again to another directory/disk, and rerun.  (The DATA segment issue seldom causes trouble for me.)  The ability to rerun programs is so important to me that I want to add more code to the libraries, that will make it easier (the stdio destructors do cause trouble; they change the default device number).